gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/FieldValue.java

```diff
@@ -0,0 +1,236 @@
+   *     Returns {@link Kind#RECORD} if the corresponding field has
+   *     {@link Field.Type#record(Field...)} type.
+   */
+  public Kind kind() {
+    return kind;
+  }
+
+  /**
+   * Return this field's value as an {@link Object}.
+   */
+  public Object value() {
+    return value;
+  }
+
+  /**
+   * Return this field's value as a {@link String}. This method should only be used if the
+   * corresponding field has primitive type ({@link Field.Type#bool()}, {@link Field.Type#string()},
+   * {@link Field.Type#floatingPoint()}, {@link Field.Type#integer()},
+   * {@link Field.Type#timestamp()}).
+   *
+   * @throws ClassCastException if the field has not primitive type
+   */
+  @SuppressWarnings("unchecked")
+  public String stringValue() {
+    return (String) value;
+  }
+
+  /**
+   * Returns this field's value as a {@link Long}. This method should only be used if the
+   * corresponding field has {@link Field.Type#integer()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws NumberFormatException if the field's value could not be converted to {@link Integer}
+   */
+  @SuppressWarnings("unchecked")
+  public long longValue() {
+    return Long.valueOf(stringValue());
+  }
+
+  /**
+   * Returns this field's value as a {@link Double}. This method should only be used if the
+   * corresponding field has {@link Field.Type#floatingPoint()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws NumberFormatException if the field's value could not be converted to {@link Double}
+   */
+  @SuppressWarnings("unchecked")
+  public double doubleValue() {
+    return Double.valueOf(stringValue());
+  }
+
+  /**
+   * Returns this field's value as a {@link Boolean}. This method should only be used if the
+   * corresponding field has {@link Field.Type#bool()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws IllegalStateException if the field's value could not be converted to {@link Boolean}
+   */
+  @SuppressWarnings("unchecked")
+  public boolean booleanValue() {
+    String stringValue = stringValue();
+    checkState(stringValue.equalsIgnoreCase("true") || stringValue.equalsIgnoreCase("false"),
+        "Field value is not of boolean type");
+    return Boolean.parseBoolean(stringValue);
+  }
+
+  /**
+   * Returns this field's value as a {@link Long}, representing a timestamp in microseconds. This
+   * method should only be used if the corresponding field has {@link Field.Type#timestamp()} type.
+   *
+   * @throws ClassCastException if the field has not primitive type
+   * @throws NumberFormatException if the field's value could not be converted to {@link Long}
+   */
+  @SuppressWarnings("unchecked")
+  public long timestampValue() {
+    return new Double(((Double.valueOf(stringValue())) * MICROSECONDS)).longValue();
```

Comment:
They are returned as a double value since epoch in the form `1408452095.22` where the integer part is **seconds**: `1408452095.22 == 2014-08-19 07:41:35.220 -05:00`